tracer/src/Datadog.Trace/ClrProfiler/AutoInstrumentation/AWS/SQS/ReceiveMessageIntegration.cs

```diff
@@ -40,21 +44,25 @@ public class ReceiveMessageIntegration
         /// <param name="request">The request for the SQS operation</param>
         /// <returns>Calltarget state value</returns>
         internal static CallTargetState OnMethodBegin<TTarget, TReceiveMessageRequest>(TTarget instance, TReceiveMessageRequest request)
-            where TReceiveMessageRequest : IAmazonSQSRequestWithQueueUrl, IDuckType
+            where TReceiveMessageRequest : IAmazonSQSRequestWithQueueUrl, IContainsMessageAttributeNames, IDuckType
         {
             if (request.Instance is null)
             {
                 return CallTargetState.GetDefault();
             }
 
+            var queueName = AwsSqsCommon.GetQueueName(request.QueueUrl);
```

Comment:
In this case, there was no point doing extra work if we didn't need to.

However, you have to be _very_ careful with nulls whenever you're working with ducktyping. The type checker does _not_ help you in most cases, because _you_ define whether things can be null or not in your ducktypes, declaratively.

That's why we added all the extra null checks (recently), because we literally saw null reference exceptions in the logs 😄  https://github.com/DataDog/dd-trace-dotnet/pull/4860